analyzers/tests/SonarAnalyzer.TestFramework.Test/Verification/VerifierTest.cs

```diff
@@ -261,6 +261,77 @@ public void Compile_Razor_NoReferences()
         references.Should().NotContain(x => x.Display.Contains("Microsoft.Azure.DocumentDB"));
     }
 
+    [TestMethod]
+    public void Compile_Razor_Snippet_NoName()
+    {
+        var compilation = DummyWithLocation
+            .AddPaths("Dummy.cshtml")
+            .AddSnippet("class Sample {}")
+            .WithLanguageVersion(LanguageVersion.Latest)
+            .Build()
+            .Compile(false)
+            .Single();
+
+        compilation.AdditionalSourceFiles.Should().ContainSingle(x => x.EndsWith("Dummy.cshtml"));
+        compilation.Compilation.SyntaxTrees.Select(x => x.FilePath).Should().Contain(x => x.EndsWith("snippet.0.cs"));
```

Comment:
These four tests are testing that:
- If a snippet name ends with `.cshtml`, they are added as additional file to the compilation, to be processed by the razor generators.
- If they either do not have a name or their name ends with `.cs`, they are **not** added as additional file, but they are still part of the compilation as a SyntaxTree.